### PR TITLE
Validations specs

### PR DESCRIPTION
**Pull request by medaminedev66**
- Add the validations to the User model
- Add the validations to  the Post model
- Add unit specs for all of the models' validations.
- Add flash messages in the create actions in all controllers.
- Make sure that the N+1 problem is solved when fetching all posts.

### CODE REVIEW

**Review by mke2111 — CHANGES_REQUESTED** [review]
### Hey @medaminedev66, 

**This is a great project you have going on, well-done lad**. 

   
**However, there are a few sections to improve before your project is approved.** 🔴 
<hr>

## Required Changes ♻️
_Check the inline comments under the review for sections to improve._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/like_spec.rb [hunk 1]
```diff
@@ -1,5 +1,22 @@
...
+  it 'update the likes counter of post' do
+    # subject.update_likes_counter
+    expect(subject.post_id).to eq(7)
+  end
+
+  it 'checks that the post_id is integer' do
+    # subject.update_likes_counter
+    subject.post_id = 7.5
+    expect(subject).to_not be_valid
+  end
```
Comment (mke2111) [inline spec/models/like_spec.rb]: - [ ] Remove the commented-out pieces of code. It is a standard practice to commit code without commented-out pieces of code.

Comment (medaminedev66) [reply spec/models/like_spec.rb]: Hi @mke2111, Thanks for reviewing my pull request. Ok, I will remove them.

**Review by mke2111 — APPROVED** [review]
### Hi @medaminedev66, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
**Congratulations! 🎉** 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**